lua/lspconfig/configs/nextflow_ls.lua

```diff
@@ -0,0 +1,37 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'java', '-jar', 'nextflow-language-server-all.jar' },
+    filetypes = { 'nextflow' },
+    root_dir = function(fname)
+      return util.root_pattern('nextflow.config')(fname) or util.find_git_ancestor(fname)
```

Comment:
```suggestion
      return util.root_pattern({ 'nextflow.config', '.git' })(fname)
```

Reply:
Will this consider priority? If there is a `nextflow.config` further up in the directory structure will it identify that as the project root before the `.git` folder? That was the main goal of this being separated

Reply:
can simply `    root_dir = util.root_pattern('nextflow.config', '.git'),`  order by index. if found  1 will stop.

Reply:
Ah, okay that's great. So just to make sure I follow, if I have this directory structure:

```
project1/
- nextflow.config
- some_folder/
  - .git/
  - test_file
```

And use the revised block while editing `test_file`, I'll get a root at `project1` and _not_ `some_folder`?

Reply:
yep it just for iparis range from index 1 if 1 found return otherwise continue to search

Reply:
Oh that's awesome! I will revise the commit with this improvement and also fix the commit history messages in a couple hours.

Thanks again for your help!

Reply:
Okay I pushed an update with this. I'm still not at a computer but I see now why the GitHub UI got the wrong commit message. It looks like you committed the suggestion I made before I was able to type out the commit message 😅 I think this PR is ready to go if you could revise your commit message for the tests can pass.


Thanks again!